src/transformers/trainer.py

```diff
@@ -4474,6 +4475,8 @@ def push_to_hub(
                 Whether the function should return only when the `git push` has finished.
             token (`str`, *optional*, defaults to `None`):
                 Token with write permission to overwrite Trainer's original args.
+            revision (`str`, *optional*):
+                The git revision to commit from. Defaults to the head of the "main" branch.
```

Comment:
Isn't it the branch to commit to - rather than the commit to commit from?

Reply:
I think this description comes from the [hub upload_folder api](https://github.com/huggingface/huggingface_hub/blob/6f49af5051bca567798385b673af76642adf4cf6/src/huggingface_hub/hf_api.py#L4823). I was only going to use it for branches and tested with that, so not sure if it works for other commit hashes.

Reply:
OK!